engine/src/main/java/org/terasology/logic/debug/MovementDebugCommands.java

```diff
@@ -233,28 +238,59 @@ public String stepHeight(@Sender EntityRef client, @CommandParam("height") float
-            if (move != null) {
-                float prevHeight = move.height;
-                move.height = amount;
-                clientComp.character.saveComponent(move);
-                LocationComponent loc = client.getComponent(LocationComponent.class);
-                Vector3f currentPosition = loc.getWorldPosition();
-                clientComp.character
-                        .send(new CharacterTeleportEvent(new Vector3f(currentPosition.getX(), currentPosition.getY() + (amount - prevHeight) / 2, currentPosition.getZ())));
-                physics.removeCharacterCollider(clientComp.character);
-                physics.getCharacterCollider(clientComp.character);
-                return "Height of player set to " + amount + " (was " + prevHeight + ")";
+        if (amount >= 1 && amount <= 25) {
+            try {
+                ClientComponent clientComp = client.getComponent(ClientComponent.class);
+                CharacterMovementComponent move = clientComp.character.getComponent(CharacterMovementComponent.class);
+                CharacterComponent charComp = clientComp.character.getComponent(CharacterComponent.class);
+                EntityRef player = clientComp.character;
+                GazeMountPointComponent gazeMountPointComponent = player.getComponent(GazeMountPointComponent.class);
+                Optional<Prefab> prefab = Assets.get(new ResourceUrn("engine:player"), Prefab.class);
+                CharacterMovementComponent moveDefault = prefab.get().getComponent(CharacterMovementComponent.class);
+
+                if (move != null && gazeMountPointComponent != null) {
+                    float prevHeight = move.height;
+                    float defaultEyeHeight = config.getPlayer().getEyeHeight();
+                    float foreHeadSize = moveDefault.height - defaultEyeHeight;
+                    float ratio = amount / moveDefault.height;
+
+                    move.height = amount;
+                    move.jumpSpeed = getJumpSpeed(ratio);
+                    move.stepHeight = moveDefault.stepHeight * ratio;
+                    move.distanceBetweenFootsteps = moveDefault.distanceBetweenFootsteps * ratio;
+                    move.runFactor = getRunFactor(ratio);
+                    charComp.interactionRange = getInteractionRange(ratio);
+                    gazeMountPointComponent.translate.y = amount - foreHeadSize;
+                    Location.removeChild(player, gazeMountPointComponent.gazeEntity);
+                    Location.attachChild(player, gazeMountPointComponent.gazeEntity, gazeMountPointComponent.translate, new Quat4f(Quat4f.IDENTITY));
+
+                    player.saveComponent(gazeMountPointComponent);
+                    clientComp.character.saveComponent(move);
+
+                    return "Height of player set to " + amount + " (was " + prevHeight + ")";
+                }
+                return "";
+            } catch (NullPointerException e) {
```

Comment:
If we did proper null checks this try-catch block should not be necessary (in general, catching a NPE is always a code smell as you could have done a null-check before).